### PR TITLE
add developers guide

### PR DESCRIPTION
**Pull request by MouceL**

### CODE REVIEW

File: doc/zh/slime-developers-guide.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+|   `-- module.go
+`-- publish.sh
+```
+
+其中的一些目录含义如下：
+
+- api：
+  - config：子模块的启动参数定义，例如，limiter定义的disableGlobalRateLimit字段，在启动时根据该参数关闭全局共享限流功能
+  - v1alpha1：子模块的CRD定义，默认情况下生成的CRD的gvk是`microservice.slime.io/v1alpha1/Foo`,其中group和version固定不变，kind等同于用户的子模块名称
+- controller: 核心的controller逻辑，开发这需要关注的目录
```
Comment (cywang1905) [inline doc/zh/slime-developers-guide.md:57]: 开发者

File: doc/zh/slime-developers-guide.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+	// Foo2 is an foo field of Foo.
+	Foo2 string `protobuf:"bytes,2,opt,name=foo2,proto3" json:"foo2,omitempty"`
+	// Foo3 is an foo field of Foo.
+	Foo3                 string   `protobuf:"bytes,3,opt,name=foo3,proto3" json:"foo3,omitempty"`
+	XXX_NoUnkeyedLiteral struct{} `json:"-"`
+	XXX_unrecognized     []byte   `json:"-"`
+	XXX_sizecache        int32    `json:"-"`
+}
+```
+**note: proto中的oneof类型目前不能优雅支持，请使用者谨慎使用**
```
Comment (cywang1905) [inline doc/zh/slime-developers-guide.md:133]: 再说清晰点？能用的话，给个样例，有什么已知的坑

Comment (MouceL) [reply doc/zh/slime-developers-guide.md:133]: 不能用

File: staging/src/slime.io/slime/modules/foo/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Use distroless as minimal base image to package the manager binary
```
Comment (YonkaFang) [inline staging/src/slime.io/slime/modules/foo/Dockerfile]: this module (name) will confuse the users

File: Makefile [hunk 4]
```diff
@@ -47,6 +47,7 @@ $(MODULE_ROOTS):
 	@echo "generate k8s object for module $(notdir $@)"
 	@pushd $@ 1>/dev/null 2>&1; \
 	controller-gen object:headerFile="$(GO_HEADER_FILE)" paths="./api/..."; \
+	controller-gen crd paths="./api/..." output:crd:dir="./charts/crds"; \
```
Comment (YonkaFang) [inline Makefile:50]: these two commits seem to can be squashed

File: staging/src/slime.io/slime/modules/foo/Dockerfile (hunk 3 above)
Comment (MouceL) [reply staging/src/slime.io/slime/modules/foo/Dockerfile]: removed

File: Makefile (hunk 4 above)
Comment (MouceL) [reply Makefile:50]: update

**Pull request merged**